subsys/net/lib/sockets/socketpair.c

```diff
@@ -550,10 +550,7 @@ static ssize_t spair_write(void *obj, const void *buffer, size_t count)
 		}
 	}
 
-	res = k_pipe_put(&remote->recv_q, (void *)buffer, count,
-			 &bytes_written, 1, K_NO_WAIT);
-	__ASSERT(res == 0, "k_pipe_put() failed: %d", res);
-
+	bytes_written = ring_buf_put(&remote->recv_q, (void *)buffer, count);
```

Comment:
I know it's claimed so in a comment, but can you be explicit about the synchronization in the commit message.  Basically: k_pipe is a threadsafe synchronized tool, while ring_buf is a raw data structure and isn't.  I just want to see a statement to the effect of "All the locations being changed here are already protected by this spinlock (or semaphore, or whatever)" and don't need further locking.

Reply:
I updated the commit message accordingly :+1: